prov/efa/src/efa_cq.h

```diff
@@ -3,6 +3,77 @@
 
 #include "efa.h"
 
+enum ibv_cq_ex_type {
+	IBV_CQ,
+	EFADV_CQ
+};
+
+struct efa_ibv_cq {
+	struct ibv_cq_ex *ibv_cq_ex;
+	enum ibv_cq_ex_type ibv_cq_ex_type;
+};
+
+struct efa_ibv_cq_poll_list_entry {
```

Comment:
what's the advantage of this separate entry struct instead of adding `dlist_entry` to `efa_ibv_cq`?

Reply:
Because the same efa_ibv_cq can be added to different lists (tx cq's poll list and rx cq's poll list). I find if I add a dlist_entry in efa_ibv_cq and make it be inserted directly, it will mess up and cause data corruption

Reply:
OK instead of adding multiple dlist_entry to efa_ibv_cq, you create a separate struct here. Makes sense.